can/interfaces/remote/events.py

```diff
@@ -0,0 +1,451 @@
+"""
+Different events can be sent and transmitted over the network connection.
+
+Examples:
+  * Messages
+  * Exceptions
+  * Transmit success
+  * Transmit failure
+  * ...
+"""
+
+import struct
+import logging
+import can
+
+logger = logging.getLogger(__name__)
+
+
+class BaseEvent(object):
+    """Events should inherit this class."""
+
+    def encode(self):
+        """Convert event data to bytes.
+
+        :return:
+            Bytestring representing the event data.
+        """
+        return b''
+
+    @classmethod
+    def from_buffer(cls, buf):
+        """Parse the data and return a new event.
+
+        :param buf:
+            Bytestring representing the event data.
+
+        :return:
+            Number of bytes actually processed.
+
+        :raise: :class:`can.interfaces.remote.events.NeedMoreDataError`
+            if not enough data exists.
+        """
+        return cls()
+
+    def __len__(self):
+        return len(self.encode())
+
+    # Useful for tests
+    def __eq__(self, other):
+        return (self.__class__ == other.__class__ and
+                self.__dict__ == other.__dict__)
+
+
+class BusRequest(BaseEvent):
+    """Request for connecting to CAN bus.
+
+    +--------+-------+--------------------------------------------------------+
+    | Byte   | Type  | Contents                                               |
+    +========+=======+========================================================+
+    | 0      | U8    | Protocol version used by client                        |
+    +--------+-------+--------------------------------------------------------+
+    | 1 - 4  | S32   | Bitrate in bits/s requested                            |
+    +--------+-------+--------------------------------------------------------+
+    """
+
+    #: Event ID
+    EVENT_ID = 1
+
+    _STRUCT = struct.Struct('>Bl')
+
+    def __init__(self, version, bitrate):
+        """
+        :param int version:
+            Network protocol version
+        :param int bitrate:
+            Bitrate to use on CAN
+        """
+        #: Network protocol version
+        self.version = version
```

Comment:
The version is constant (can.interfaces.remote.PROTOCOL_VERSION) so it might be unnecessary to have it in the constructor. It was more for consistency.